discover/discover.go

```diff
@@ -0,0 +1,85 @@
+package discover
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/discover/aws"
+	"github.com/hashicorp/discover/azure"
+	"github.com/hashicorp/discover/gce"
+)
+
+func Discover(cfg string, l *log.Logger) ([]string, error) {
```

Comment:
May want to consider making discoverer an interface an having a list of them that this default Discover method will lookup against.

You can see the pattern here: https://github.com/hashicorp/go-getter/blob/master/get.go#L25-L63.

This has several benefits, projects can add discoverers without upstreaming them, projects can disable discoverers, parsing of configs can live with the discoverer, etc.